### PR TITLE
Make deprecated task container methods an error

### PR DESCRIPTION
**Pull request by ghale**
Makes the following an error:

- `TaskContainer.add()`
- `TaskContainer.addAll()`
- `TaskContainer.remove()`
- `TaskContainer.removeAll()`
- `TaskContainer.retainAll()`
- `TaskContainer.clear()`
- `TaskContainer.iterator().remove()`
- Replacing a task that has already been realized.
- Replacing a registered (unrealized) task with a task that has an incompatible type (i.e. a type that is not the same type or a sub-type of the previously registered type)
- Attempting to replace a task that has not been previously been registered.

### CODE REVIEW

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputHistoryLossIntegrationTest.groovy [hunk 1]
```diff
@@ -579,52 +579,6 @@ class StaleOutputHistoryLossIntegrationTest extends AbstractIntegrationSpec {
         targetFile2.assertDoesNotExist()
     }
 
-    def "task is renamed"() {
```
Comment (big-guy) [inline subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputHistoryLossIntegrationTest.groovy]: Can we keep this but use register and replace instead?

File: subprojects/docs/src/docs/userguide/upgrading_version_5.adoc [hunk 2]
```diff
@@ -109,6 +109,24 @@ The deprecated announce and build-announcements plugins have been removed.  Ther
...
+- `TaskContainer.removeAll()`
+- `TaskContainer.retainAll()`
+- `TaskContainer.clear()`
+- `TaskContainer.iterator().remove()`
+
+Additionally, the following deprecated functionality now results in an error:
+
+- Replacing a task that has already been realized.
+- Replacing a registered (unrealized) task with a task that has an incompatible type (i.e. a type that is not the same type or a sub-type of the previously registered type)
+- Attempting to replace a task that has not been previously been registered.
```
Comment (big-guy) [inline subprojects/docs/src/docs/userguide/upgrading_version_5.adoc]:
```suggestion
- Replacing a task that was never registered.
```

File: subprojects/docs/src/docs/userguide/upgrading_version_5.adoc [hunk 3]
```diff
@@ -109,6 +109,24 @@ The deprecated announce and build-announcements plugins have been removed.  Ther
...
+- `TaskContainer.remove()`
+- `TaskContainer.removeAll()`
+- `TaskContainer.retainAll()`
+- `TaskContainer.clear()`
+- `TaskContainer.iterator().remove()`
+
+Additionally, the following deprecated functionality now results in an error:
+
+- Replacing a task that has already been realized.
+- Replacing a registered (unrealized) task with a task that has an incompatible type (i.e. a type that is not the same type or a sub-type of the previously registered type)
```
Comment (big-guy) [inline subprojects/docs/src/docs/userguide/upgrading_version_5.adoc]:
```suggestion
- Replacing a registered (unrealized) task with an incompatible type. A compatible type is the same type or sub-type of the registered type.
```

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputHistoryLossIntegrationTest.groovy (hunk 1 above)
Comment (ghale) [reply subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputHistoryLossIntegrationTest.groovy]: Yep.  Done.

File: subprojects/docs/src/docs/userguide/upgrading_version_5.adoc (hunk 2 above)
Comment (ghale) [reply subprojects/docs/src/docs/userguide/upgrading_version_5.adoc]: Fixed.

File: subprojects/docs/src/docs/userguide/upgrading_version_5.adoc (hunk 3 above)
Comment (ghale) [reply subprojects/docs/src/docs/userguide/upgrading_version_5.adoc]: Fixed

**Pull request merged**